testing/trino-tests/src/test/java/io/trino/execution/resourcegroups/db/TestQueuesDb.java

```diff
@@ -306,6 +330,26 @@ public void testQueryExecutionTimeLimit()
         waitForQueryState(queryRunner, secondQuery, FAILED);
     }
 
+    @Test(timeOut = 60_000)
+    public void testQueryPlanningTimeLimit()
+            throws Exception
+    {
+        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+        QueryId query = createQuery(
+                queryRunner,
+                testSessionBuilder()
+                        .setCatalog("tpch")
+                        .setSchema("sf100000")
+                        .setSource("dashboard")
+                        .setSystemProperty(QUERY_MAX_PLANNING_TIME, "1ms")
+                        .build(),
+                LONG_PLANNING_QUERY);
```

Comment:
Use Mock connector with some operation invoked during planning doing a sleep.

Also, planning time limit doesnt belong to TestQueues, does it?